### ISSUE TITLE
Build failure: linuxKernel.packages.linux_6_12.digimend

### ISSUE THREAD

**Issue by KCN-judu** [NixOS/nixpkgs#363873]
## Steps To Reproduce

1. build *linuxKernel.packages.linux_6_12.digimend*

## Build log

```
building the system configuration...
error: builder for '/nix/store/54fn0xwsmspfx9r4swwgc8bm22bgazcv-digimend-13.drv' failed with exit code 2;
       last 22 log lines:
       > Running phase: unpackPhase
       > unpacking source archive /nix/store/rdgi30a0zgyhwqhh2ni2i144kj4y26wh-source
       > source root is source
       > Running phase: patchPhase
       > Running phase: updateAutotoolsGnuConfigScriptsPhase
       > Running phase: configurePhase
       > no configure script, doing nothing
       > Running phase: buildPhase
       > build flags: SHELL=/nix/store/0irlcqx2n3qm6b1pc9rsd2i8qpvcccaj-bash-5.2p37/bin/bash O=\$\(buildRoot\) CC=/nix/store/3ix5h74n7ar9950vwzp4dxmil70pmx0k-gcc-wrapper-13.3.0/bin/cc HOSTCC=/nix/store/3ix5h74n7ar9950vwzp4dxmil70pmx0k-gcc-wrapper-13.3.0/bin/cc HOSTLD=/nix/store/9bacph866qhmr2zfib1h49jixq5hhd01-binutils-wrapper-2.43.1/bin/ld ARCH=x86_64 KVERSION=6.12.0 KDIR=/nix/store/lacdcr5zi7w59f8xsq4cfi41mzgc6dcn-linux-6.12-dev/lib/modules/6.12.0/build DESTDIR=/nix/store/nmxp5blxhnsldp6lipfcfssjqqi0a28s-digimend-13 INSTALL_MOD_PATH=/nix/store/nmxp5blxhnsldp6lipfcfssjqqi0a28s-digimend-13
       > make -C /nix/store/lacdcr5zi7w59f8xsq4cfi41mzgc6dcn-linux-6.12-dev/lib/modules/6.12.0/build M=/build/source modules
       > make[1]: Entering directory '/nix/store/lacdcr5zi7w59f8xsq4cfi41mzgc6dcn-linux-6.12-dev/lib/modules/6.12.0/build'
       >   CC [M]  /build/source/hid-kye.o
       > /build/source/hid-kye.c:790:25: error: initialization of 'const __u8 * (*)(struct hid_device *, __u8 *, unsigned int *)' {aka 'const unsigned char * (*)(struct hid_device *, unsigned char *, unsigned int *)'} from incompatible pointer type '__u8 * (*)(struct hid_device *, __u8 *, unsigned int *)' {aka 'unsigned char * (*)(struct hid_device *, unsigned char *, unsigned int *)'} [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wincompatible-pointer-types-Werror=incompatible-pointer-types8;;]
       >   790 |         .report_fixup = kye_report_fixup,
       >       |                         ^~~~~~~~~~~~~~~~
       > /build/source/hid-kye.c:790:25: note: (near initialization for 'kye_driver.report_fixup')
       > cc1: some warnings being treated as errors
       > make[3]: *** [/nix/store/lacdcr5zi7w59f8xsq4cfi41mzgc6dcn-linux-6.12-dev/lib/modules/6.12.0/source/scripts/Makefile.build:229: /build/source/hid-kye.o] Error 1
       > make[2]: *** [/nix/store/lacdcr5zi7w59f8xsq4cfi41mzgc6dcn-linux-6.12-dev/lib/modules/6.12.0/source/Makefile:1936: /build/source] Error 2
       > make[1]: *** [/nix/store/lacdcr5zi7w59f8xsq4cfi41mzgc6dcn-linux-6.12-dev/lib/modules/6.12.0/source/Makefile:224: __sub-make] Error 2
       > make[1]: Leaving directory '/nix/store/lacdcr5zi7w59f8xsq4cfi41mzgc6dcn-linux-6.12-dev/lib/modules/6.12.0/build'
       > make: *** [Makefile:25: modules] Error 2
       For full logs, run 'nix log /nix/store/54fn0xwsmspfx9r4swwgc8bm22bgazcv-digimend-13.drv'.
error: 1 dependencies of derivation '/nix/store/d35s175akmjqdq83z66xqp5xs1mk4fym-system-path.drv' failed to build
error: 1 dependencies of derivation '/nix/store/hcn5166ricfm6grg3vv12gm22a2h1bjj-nixos-system-nixos-25.05.20241119.23e89b7.drv' failed to build
```

This seems to be caused by treating **incompatible-pointer-types** as an error instead of a warning during the build process.

## Metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.0, NixOS, 25.05 (Warbler), 25.05.20241119.23e89b7`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.10`
 - nixpkgs: `/nix/store/c9wv7i0af6mysmy65x6nvyfw5izzxv4g-source`

## Notify maintainers

@gebner 
---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by dramforever** [comment]
digimend appears to be unmaintained https://spbnick.github.io/2024/11/03/Letting-go-of-DIGImend.html

oops

**Comment by SohamG** [comment]
I have this exact problem, and it seems like the fix (for now) is just a one line patch addding/removing one const (which is kind of cosmetic in c/++ anyway...) or disabling warnings=errors in the CFLAGs. Wonder if the current maintainer would look into this?

**Comment by SohamG** [comment]
CC @gebner

**Comment by SohamG** [comment]
I've added the PR in upstream digimend (https://github.com/DIGImend/digimend-kernel-drivers/pull/707) that fixes this as a patch to the current Nix expression in #373869

**Comment by KCN-judu** [comment]
thanks for your reply, i'll try it later and close this issue

**Comment by kanashimia** [comment]
> With it disabled, I can build my system without this build failure. It did not break anything for my drawing tablet as far as I can tell. This may be because some of digimend has been moved into the kernel.

Hello, I've contributed to Digimend so I know some things, situation is like this:
It hasn't been maintained for a very long time, and development shifted to being in the kernel a long time ago, afterwards Digimend was basically just a downstream from the kernel drivers which you could install on an older system.
And recently development of new tablet drivers shifted to being eBPF based: https://gitlab.freedesktop.org/libevdev/udev-hid-bpf
Well, this whole situation is described in that blog post above: https://spbnick.github.io/2024/11/03/Letting-go-of-DIGImend.html

TL;DR: There is no reason to use Digimend drivers nowadays, unless you are on an ancient kernel.
They should be deprecated and removed eventually instead.

**Comment by dramforever** [comment]
Despite the comment above there are nonetheless still tablets that are not supported by mainline kernel or udev-hid-bpf. Furthermore, udev-hid-bpf isn't even packaged in nixpkgs. It seems pretty weird to say that "There is no reason to use Digimend drivers nowadays, unless you are on an ancient kernel" when to do so some users would need to figure out udev-hid-bpf, possibly even write an eBPF program just so they can use their hardware that worked with digimend.

**Comment by kanashimia** [comment]
@dramforever Tablets that are supported by udev-hid-bpf are not supported by digimend or kernel drivers, and pretty much all drivers supported by digimend are supported in the kernel, the only one that isn't supported is a single tablet by polostar, and it can be ported trivially if anyone actually has it. If you can name any driver that is supported only by digimed drivers except that one then please do so.

**Comment by dramforever** [comment]
... although, that gave me the idea to just patch and build the hid-uclogic module out of tree. i don't know how cursed it is but here, it does seem to work: https://github.com/dramforever/config/commit/b7ccb688829745b2e993153846f69b28865e3752#diff-1793ae0624995cc5460c37588debda79651f3ad754a1400edfca2295e14d25bf

**Comment by kanashimia** [comment]
@dramforever I see, thanks. Jose probably just forgot about this, it should be good for inclusion, I mailed him.
